src/LibYear.Lib/FileTypes/XmlProject.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace LibYear.Lib.FileTypes
+{
+    public abstract class XmlProject : IProjectFile
+    {
+        public virtual string FileName => throw new NotSupportedException();
+        protected readonly XDocument _xmlContents;
+        protected readonly Stream _underlyingStreamData;
+        public IDictionary<string, NuGetVersion> Packages { get; protected set; }
+
+        public XmlProject(Stream fileStream, string elementName, string packageAttributeName, string versionAttributeName)
+        {
+            _underlyingStreamData = fileStream;
+            _xmlContents = XDocument.Load(fileStream);
+
+
+            Packages = _xmlContents.Descendants(elementName)

```

Comment:
this is a really, hard to read line. Could this be improved with private extension methods? or split up? Yes, it technically can all be one line .. but maintainability must be paramount also.